src/sciline/param_table.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from typing import Any, Collection, Dict, Mapping, Optional
```

Comment:
`typing.Collection` is deprecated in favour of `collections.abc.Collection` in python 3.9

Reply:
My problem is that `typing.abc.* is not generic in 3.8.